File: notes-n-resources/Data-Structures-N-Algo/_DS-n-Algos/__MY_OPRIGINAL_DS/_Extra-Practice/05_hash_tables/c++11/01_price_of_groceries.cpp
#include <iostream>
#include <unordered_map>
#include <string>
#include <utility>

using std::cout;
using std::endl;

int main() {
    std::unordered_map<std::string, float> book = {
        {"apple", 0.67},
        {"milk", 1.49},
        {"avocado", 1.49}
    };

    // print book
    for (std::pair<std::string, float> pair : book) {
        cout << pair.first << ": " << pair.second << "$" << endl;
    }
}

File: notes-n-resources/Data-Structures-N-Algo/_DS-n-Algos/__MY_OPRIGINAL_DS/_Extra-Practice/05_hash_tables/ts/01_price_of_groceries.ts
interface HashTable<T> {
    [key: string]: T;
}

const book: HashTable<number> = {};

book['apple'] = 0.67;
book['milk'] = 1.49;
book['avocado'] = 1.49;

console.log(book); // { apple: 0.67, milk: 1.49, avocado: 1.49 }